cpp/ql/src/Likely Bugs/Leap Year/UncheckedReturnValueForTimeFunctions.ql

```diff
@@ -0,0 +1,109 @@
+    }
+}
+
+/**
+ * This list of APIs should check for the return value to detect problems durin gthe conversion
+ */
+class TimeConversionFunction extends Function {
+    TimeConversionFunction() {
+        this.getQualifiedName().matches("FileTimeToSystemTime")
+        or this.getQualifiedName().matches("SystemTimeToFileTime")
+        or this.getQualifiedName().matches("SystemTimeToTzSpecificLocalTime")
+        or this.getQualifiedName().matches("SystemTimeToTzSpecificLocalTimeEx")
+        or this.getQualifiedName().matches("TzSpecificLocalTimeToSystemTime")
+        or this.getQualifiedName().matches("TzSpecificLocalTimeToSystemTimeEx")
+        or this.getQualifiedName().matches("RtlLocalTimeToSystemTime")
+        or this.getQualifiedName().matches("RtlTimeToSecondsSince1970")
+        or this.getQualifiedName().matches("_mkgmtime")
+    }
+}
+
+from FunctionCall fcall, TimeConversionFunction trf
+    , Variable var
+where fcall = trf.getACallToThisFunction()
+    and fcall instanceof ExprInVoidContext
+    and var.getUnderlyingType() instanceof DateDataStruct
+    and ( exists( AddressOfExpr aoe | 
+          aoe = fcall.getAnArgument()
+          and aoe.getAddressable() = var 
+        ) or exists( VariableAccess va |
+            fcall.getAnArgument() = va
+            and var.getAnAccess() = va
+        )
+    )
+    and exists( DateStructModifiedFieldAccess dsmfa, VariableAccess modifiedVarAccess |
+         modifiedVarAccess = var.getAnAccess() 
+        and modifiedVarAccess  = dsmfa.getQualifier()
+         and modifiedVarAccess = fcall.getAPredecessor*()
+    )
+    // Remove false positives
+    and not 
+    (
+    // Remove any instance where the predecessor is a SafeTimeGatheringFunction and no change to the data happened in between
+    exists( FunctionCall pred |
+        pred = fcall.getAPredecessor*()
+        and exists( SafeTimeGatheringFunction stgf | 
+            pred = stgf.getACallToThisFunction()
+        )
+        and not exists( DateStructModifiedFieldAccess dsmfa, VariableAccess modifiedVarAccess |
+             modifiedVarAccess = var.getAnAccess() 
+             and modifiedVarAccess  = dsmfa.getQualifier()
+             and modifiedVarAccess = fcall.getAPredecessor*()
+             and modifiedVarAccess = pred.getASuccessor*()
+        )
+    )
+    // Remove any instance where the year is changed, but the month is set to 1 (year wrapping)
+    or 
+    exists(MonthFieldAccess  mfa, AssignExpr ae |
+            mfa.getQualifier() = var.getAnAccess()
+            and mfa.isModified()
+            and mfa = fcall.getAPredecessor*()
+            and ae = mfa.getEnclosingElement()
+            and ae.getAnOperand().getValue().toInt() = 1 
+        )
+    )
+select fcall, "Return value of $@ function should be checked to verified for any error because variable $@ is not guaranteed to be safe.", trf, trf.getQualifiedName().toString(), var, var.getName()
```

Comment:
"should be checked to verified for any error"
I'm not sure what this should be.
Maybe:
"should be verified to check for any error"